src/widget.h

```diff
@@ -271,7 +278,8 @@ class widget
         // Layout this widget within max_width, including child widgets. Calling layout on a regular
         // (non-layout style) widget is the same as show(), but will pad with spaces inside the
         // label area, so the returned string is equal to max_width.
-        std::string layout( const avatar &ava, unsigned int max_width = 0, int label_width = 0 );
+        std::string layout( const avatar &ava, std::string &_label_separator, unsigned int max_width = 0,
```

Comment:
You could make `_label_separator` an optional parameter at the end of the list, that way you wouldn't have to modify the calls to layout() in the tests

```diff
-       std::string layout( const avatar &ava, std::string &_label_separator, unsigned int max_width = 0,
-                           int label_width = 0 );
+       std::string layout( const avatar &ava, unsigned int max_width = 0, int label_width = 0,
+                           const std::string &_label_separator = "DEFAULT" );
```

Reply:
Issue is if I do that, the tests fail because the default value for the separator is the rather useless `DEFAULT`. So the widget for strength would look like this: `STRDEFAULT8`, which is... not ideal.

Reply:
I'd add `"separator": ": "` to the widgets in data/mods/TEST_DATA/widgets.json, that way you don't have to modify the test code and it also acts as a test for your new field

Reply:
Actually, it looks like the `label_separator` argument is completely unnecessary. I'm going over the code again to see if I can take it out.